src/dms/frontend/EdFi.DataManagementService.Frontend.AspNetCore/Modules/TokenEndpointModule.cs

```diff
@@ -17,28 +17,37 @@ public void MapEndpoints(IEndpointRouteBuilder endpoints)
         endpoints.MapPost("/oauth/token", GenerateToken);
     }
 
-    internal static async Task GenerateToken(HttpContext httpContext, IOptions<AppSettings> appSettings, IOAuthManager oAuthManager, ILogger<TokenEndpointModule> logger, IHttpClientFactory httpClientFactory)
+    internal static async Task GenerateToken(
+        HttpContext httpContext,
+        IOptions<AppSettings> appSettings,
+        IOAuthManager oAuthManager,
+        ILogger<TokenEndpointModule> logger,
+        IHttpClientFactory httpClientFactory
+    )
     {
-        // Create client for sending upstream request.
-        logger.LogInformation("Handling OAuth Token request.");
-        var client = httpClientFactory.CreateClient();
+        var traceId = AspNetCoreFrontend.ExtractTraceIdFrom(httpContext.Request, appSettings);
+        logger.LogInformation(
+            "Received token request for forwarding to identity provider - {TraceId}",
+            traceId
+        );
+
+        var client = new HttpClientWrapper(httpClientFactory.CreateClient());
 
-        // Extract the Authorization Headers.
         httpContext.Request.Headers.TryGetValue("Authorization", out var authHeader);
-        try
-        {
-            logger.LogInformation("Requesting access token from OAuth Manager.");
-            var response = await oAuthManager.GetAccessTokenAsync(client, authHeader.ToString(), appSettings.Value.AuthenticationService);
 
-            logger.LogInformation("OAuth Manager access token request successful.");
-            await response.Content.CopyToAsync(httpContext.Response.Body);
-        }
-        catch (OAuthIdentityException ex)
+        var response = await oAuthManager.GetAccessTokenAsync(
+            client,
+            authHeader.ToString(),
+            appSettings.Value.AuthenticationService,
+            traceId
+        );
+
+        if (logger.IsDebugEnabled())
         {
-            logger.LogError(ex, "Error from OAuthManager service");
-            httpContext.Response.StatusCode = (int?)ex.StatusCode ?? (int)HttpStatusCode.BadGateway;
-            await httpContext.Response.WriteAsync($"Error Getting Access Token Async: {ex.Message}");
+            logger.LogDebug("Upstream token response code {Code} - {TraceId}", response.StatusCode, traceId);
```

Comment:
Sometimes on exit from a function I added debug logging to help with troubleshooting. Note the `logger.IsDebugEnabled()` function call. We use that because sometimes the debug logging requires some extra processing, and we want to avoid that processing if the message will just be thrown away. That is not the case here, but I still used the guard clause just to "obey the rules."